libs/server/Resp/Parser/SessionParseState.cs

```diff
@@ -158,10 +169,26 @@ public void InitializeWithArguments(ArgSlice[] args)
 
             for (var i = 0; i < args.Length; i++)
             {
-                buffer[i] = args[i];
+                *(bufferPtr + i) = args[i];
             }
         }
 
+        /// <summary>
+        /// Limit access to the argument buffer to start at a specified index
+        /// and end after a specified number of arguments.
+        /// </summary>
+        /// <param name="idxOffset">Offset value to the underlying buffer</param>
+        /// <param name="count">Argument count (default: -1 for all remaining arguments)</param>
+        public void Slice(int idxOffset, int count = -1)
```

Comment:
Should return the new SPS.
Shoud not have to incur the overhead of array lookup + pointer conversion.
Instead, it can do: return new SPS(bufferPtr + start, etc)